src/libostree/ostree-repo-commit.c

```diff
@@ -2237,14 +2237,24 @@ _ostree_repo_commit_modifier_apply (OstreeRepo               *self,
   OstreeRepoCommitFilterResult result;
   GFileInfo *modified_info;
 
-  if (modifier == NULL || modifier->filter == NULL)
+  if (modifier == NULL ||
+      (modifier->filter == NULL &&
+       (modifier->flags & OSTREE_REPO_COMMIT_MODIFIER_FLAGS_CANONICAL_PERMISSIONS) == 0))
     {
       *out_modified_info = g_object_ref (file_info);
       return OSTREE_REPO_COMMIT_FILTER_ALLOW;
     }
 
   modified_info = g_file_info_dup (file_info);
   result = modifier->filter (self, path, modified_info, modifier->user_data);
+
+  if ((modifier->flags & OSTREE_REPO_COMMIT_MODIFIER_FLAGS_CANONICAL_PERMISSIONS) != 0 &&
+      g_file_info_get_file_type (file_info) == G_FILE_TYPE_REGULAR)
+    {
+      guint current_mode = g_file_info_get_attribute_uint32 (modified_info, "unix::mode");
+      g_file_info_set_attribute_uint32 (modified_info, "unix::mode", current_mode | 0744);
```

Comment:
If you're committing to a bare-user-only repo, those are going to be discarded.

I can't think of a scenario where one is committing to an archive repo and wants to use this flag, but *doesn't* also want the uid/gid/no-xattrs.

I get what you're saying about everything else being achievable, but the flip side is - if we're not aware of a use case, why add it?

(A small reason to go the flag = full canon route is that at some point I'd like to optimize commit so we don't `malloc()` like there's no tomorrow, and part of that is going to be to stop allocating `GFileInfo` in favor of a `struct stat` on the stack.  Which we can only do right now if we don't have a commit modifier)